squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java

```diff
@@ -219,47 +175,95 @@ public HtmlScraper() {
 				updatedList.add(t);
 			}
 		}
-		
-	
-    	
+
+
+
     	return updatedList;
     }
-       
 
-    private Set<Triple> scrapeDownloadLink(Map<String, Object> resources, File htmlFile, String uri) throws Exception {
-        this.doc = Jsoup.parse(htmlFile, "UTF-8");
+    private long getJavascriptTimeout(CrawleableUri uri){
+        long timeout;
+        try {
+            if (uri.getData("time-out").equals(null)) {
+                timeout = Constants.JAVASCRIPT_WAIT_TIME;
+            }else{
+                timeout = (long) uri.getData("time-out");
+            }
+        } catch (Exception e) {
+            LOGGER.error("An error occurred when retrieving the Time out value, ", e);
+            timeout = Constants.JAVASCRIPT_WAIT_TIME;
+        }
+        return timeout;
+    }
 
-        Set<Triple> triples = new LinkedHashSet<Triple>();
+    /**
+     * Method to execute java script commands in a html page.
+     * @param uri
+     * @throws IOException
+     */
+    private void executeJavaScript(CrawleableUri uri){
+        WebClient webClient = new WebClient(BrowserVersion.FIREFOX_60);
```

Comment:
Please check whether you can reuse the `WebClient` object. At the moment, the single methods are creating new instances all the time which may waste a lot of resources.